tests/functional/relation_tests/test_relation_type_change.py

```diff
@@ -34,31 +40,86 @@ def error_message(self) -> str:
 
 relations = [
     Model(models.VIEW, "view"),
-    Model(models.TABLE, "table"),
-    Model(models.DYNAMIC_TABLE, "dynamic_table"),
+    Model(models.TABLE, "table", "default"),
+    # to be activated upon merge of dynamic table support PR
+    # Model(models.DYNAMIC_TABLE, "dynamic_table", "default"),
+    # Model(models.DYNAMIC_ICEBERG_TABLE, "dynamic_table", "iceberg"),
+    Model(models.ICEBERG_TABLE, "table", "iceberg"),
+    Model(models.ICEBERG_INCREMENTAL_TABLE, "table", "iceberg", incremental=True),
 ]
 scenarios = [Scenario(*scenario) for scenario in product(relations, relations)]
 
 
 class TestRelationTypeChange:
 
+    @staticmethod
+    def include(scenario) -> bool:
+        return (
+            scenario.initial.table_format != "iceberg" and scenario.final.table_format != "iceberg"
+        )
+
     @pytest.fixture(scope="class", autouse=True)
     def seeds(self):
         return {"my_seed.csv": models.SEED}
 
     @pytest.fixture(scope="class", autouse=True)
     def models(self):
-        yield {f"{scenario.name}.sql": scenario.initial.model for scenario in scenarios}
+        yield {
+            f"{scenario.name}.sql": scenario.initial.model
+            for scenario in scenarios
+            if self.include(scenario)
+        }
 
     @pytest.fixture(scope="class", autouse=True)
     def setup(self, project):
         run_dbt(["seed"])
         run_dbt(["run"])
         for scenario in scenarios:
-            update_model(project, scenario.name, scenario.final.model)
+            if self.include(scenario):
+                update_model(project, scenario.name, scenario.final.model)
         run_dbt(["run"])
 
     @pytest.mark.parametrize("scenario", scenarios, ids=[scenario.name for scenario in scenarios])
     def test_replace(self, project, scenario):
-        relation_type = query_relation_type(project, scenario.name)
-        assert relation_type == scenario.final.relation_type, scenario.error_message
+        if self.include(scenario):
+            relation_type = query_relation_type(project, scenario.name)
+            assert relation_type == scenario.final.relation_type, scenario.error_message
+            if relation_type == "dynamic_table":
```

Comment:
We're only checking `table_format` on dynamic tables, but we're including standard iceberg tables in the scenarios. I think it's worth creating a "`describe`-like" macro for tables as well to test this, given that one of the goals is to make sure we can create iceberg tables over non-iceberg tables.

The difference in parsing can be handled directly in a new `describe_table` helper in `utils`, akin to `describe_dynamic_table` (i.e. I don't think there's a need to create a whole RelationConfig for tables as there is for DTs just to test this).

Reply:
I think we should do this, but this is covered implicitly by Iceberg table tests where a swap is done.